rollup-node.md

```diff
@@ -0,0 +1,285 @@
+    the first L1 block for which an L2 block was produced).
+
+[`ForkchoiceStateV1`]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#ForkchoiceStateV1
+
+> **Note:** the properties of this `ForkchoiceStateV1` can be used to anchor queries
+> to the regular (non-engine-API) JSON-RPC endpoint of the execution engine.
+> [See here for more information.][L2-JSON-RPC-API]
+
+[L2-JSON-RPC-API]: TODO
+
+> **TODO LINK** L2 JSON RPC API (might be the same as [L1's][JSON-RPC-API])
+
+The `payloadID` returned by [`engine_forkchoiceUpdatedOPV1`] can then be passed
+to [`engine_getPayloadV1`] in order to obtain an [`ExecutionPayloadV1`], which
+fully defines a new L2 block.
+
+[`engine_getPayloadV1`]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#engine_getpayloadv1
+[`ExecutionPayloadV1`]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#executionpayloadv1
+
+## Handling L1 Re-Orgs
+
+The [previous section on L2 chain derivation][l2-chain-derivation] assumes
+linear progression of the L1 chain. It is also applicable for batch processing,
+meaning that any given point in time, the canonical L2 chain is given by
+processing the whole L1 chain since the [L2 chain inception].
+
+The L1 chain may occasionally re-organize, meaning the head of the L1 chain
+changes to a block that is not the child of the previous head but rather one of
+its "cousins" (i.e. the descendant of an ancestor of the previous head). In
+those case, the rollup driver must:
+
+1. Locate the *common ancestor*, a block that is an ancestor of both the
+   previous and new head.
+2. Isolate the range of L1 blocks \]common ancestor, ..., new head\].
+3. For each such block, call [`engine_forkchoiceUpdatedOPV1`] and
+   [`engine_getPayloadV1`].
+   - Fill the [`PayloadAttributesOPV1`] object according to [the section on payload attributes][payload-attr].
+   - Fill the [`ForkchoiceStateV1`] object according to the [the section on the
+     execution engine][calling-exec-engine], but set `headBlockHash` to the hash
+     of the last processed L2 block (use the hash of the common ancestor
+     initially) instead of the last L2 chain head. `safeBlockHash` and
+     `finalizedBlockHash` must be updated accordingly.
+
+> Note that post-[merge], the L1 chain will offer finalization guarantees
+> meaning that it won't be able to re-org more than `FINALIZATION_DELAY_BLOCKS`
+> in the past, hence preserving our finalization guarantees.
+
+> **TODO** Enunciate our finalization guarantees.
+
+> **TODO** Must enunciate error cases more precisely, as well as state what
+> should be impossible and should cause the node to stop deriving the L2 chain.
+
+## Block Gossip
+
+Another responsability of the rollup node is to transmit the L2 blocks it
+derives from the L1 chain to other L2 nodes via a process called *[block
+gossip]*.
+
+This is specified in [a separate document][gossip-spec].
```

Comment:
When linking docs we should just use the document title, instead of "separate document" etc.